From 5b049bb70d9896ef68735d2f0ce78fc28a36ad9d Mon Sep 17 00:00:00 2001
From: Juri Lelli <juri.lelli@gmail.com>
Date: Mon, 21 May 2012 14:59:35 +0200
Subject: [PATCH 26/48] sched-dl: do nothing at replenishing time if policy
 changed.

If user changed policy or policy attributes at replenish
timer firing, just do nothing (don't actually replenish).
---
 kernel/sched/dl.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/kernel/sched/dl.c b/kernel/sched/dl.c
index 367fda5..0360d4d 100644
--- a/kernel/sched/dl.c
+++ b/kernel/sched/dl.c
@@ -488,9 +488,10 @@ static enum hrtimer_restart dl_task_timer(struct hrtimer *timer)
 	/*
 	 * We need to take care of a possible races here. In fact, the
 	 * task might have changed its scheduling policy to something
-	 * different from SCHED_DEADLINE (through sched_setscheduler()).
+	 * different from SCHED_DEADLINE or changed its reservation
+	 * parameters (through sched_setscheduler()).
 	 */
-	if (!dl_task(p))
+	if (!dl_task(p) || dl_se->dl_new)
 		goto unlock;
 
 	dl_se->dl_throttled = 0;
-- 
1.7.9.5

